### PR TITLE
* [android] add timeout config and put/delete/head/patch method to fetch

### PR DESCRIPTION
**Pull request by denghongcai**
0. add timeout option to fetch options
0. add PUT/DELETE/HEAD/PATCH method support

@luics

### CODE REVIEW

**Comment by luics** [comment]
@denghongcai it's wonderful to add methods for fetch.
@sospartan pls have a review.
@acton393 we still need ios version.

File: android/sdk/src/main/java/com/taobao/weex/http/Options.java [hunk 1]
```diff
@@ -216,13 +216,23 @@
   private Map<String, String> headers;
   private String body;
   private Type type = Type.text;
+  private int timeout = DEFAULT_TIMEOUT;
```
Comment (sospartan) [inline android/sdk/src/main/java/com/taobao/weex/http/Options.java]: change to `timeoutMillis`,  I mean add 'Millis' suffix to these field/arguments.

File: android/sdk/src/main/java/com/taobao/weex/http/Options.java [hunk 2]
```diff
@@ -245,16 +255,21 @@ public Type getType() {
     return type;
   }
 
+  public int getTimeout() { return timeout; }
+
   public enum Type {
     json, text
   }
 
+  public static final int DEFAULT_TIMEOUT = 2500;
```
Comment (sospartan) [inline android/sdk/src/main/java/com/taobao/weex/http/Options.java]: Define a const value in `WXRequest` and use a reference from there.

**Comment by sospartan** [comment]
It's awesome.

**Comment by acton393** [comment]
@denghongcai   cool ,I'll check it in iOS later

**Comment by denghongcai** [comment]
@sospartan update, but I think the `timeout` concept should equal to xhr, which has a potential meaning of `Millis`

**Comment by sospartan** [comment]
The H5-render in 'android' branch is not newest, so you should move the 'H5' part of this PR to a separate one, which will merge into 'dev' branch. It's our problem, the branch management is still not working well.  
Every PR should be created according to the 'dev' branch. @luics

**Comment by luics** [comment]
@deng19891006 sorry, you should checkout branch from `dev` branch, and merge it to `dev` too.

**Comment by denghongcai** [comment]
@luics you @ the wrong guy...
the contributor guide should be updated soon...really...

this pr contains changes for both android and html5, how to proceed this? do i need to submit 2 separate pr?

**Comment by sospartan** [comment]
Send a new PR to 'dev' is all you need to do.

**Comment by luics** [comment]
@denghongcai actually, you're the first one to submit such a wonderful pr. We'll update `contributing.md` late this week after new releasing. Thanks again.

**Comment by luics** [comment]
@denghongcai new branch policy https://github.com/alibaba/weex/issues/706

**Pull request closed without merging**